### ISSUE TITLE
Re-run the citation extractor

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#4566]
For the citator project, we badly need to re-run the citation extractor. 

In today's case law entmoot, we discussed a few enhancements we should make before we do:

1. If a citation cannot be disambiguated, that's OK, we can send it to a `/c/` page, which will list multiple options.

2. If a citation cannot be matched, we can also link to it via `/c/`. Here's what you see when a citation is missing:

    ![Image](https://github.com/user-attachments/assets/12020358-8ef7-4f8c-815c-baca5cbb6821)

    We can tune that up to talk more about neutral citations and things like that, but it's surprisingly not bad already.

    I think these citation links should be red, like Wikipedia pages that haven't yet been created.

3. We need to fix any eyecite bugs that prevent good citation extraction. 

That's my brain dump. We do need to get on top of this so that we have the data we need.

**Comment by mlissner** [comment]
I added a bunch of sub-issues. We need to at least investigate these and make a decision about how important they are to do before the big extraction.

**Comment by mlissner** [comment]
4. Once @flooie lands the opinion cleanup PR, we can link to pincites via HTML anchors. We should use these so that citations link to the correct page of the decisions (if possible!)

**Comment by grossir** [comment]
Why don't we take advantange of the re-run to collect stats on what citations we are missing?
We could store the `NO_MATCH_RESOURCE` citations, and then analyze them to know which reporters / years we are missing the most, and prioritize how to get them

https://github.com/freelawproject/courtlistener/blob/f867bbd35cb00cfbb47f8c437845017f397bb0ff/cl/citations/annotate_citations.py#L57-L62

**Comment by mlissner** [comment]
Yeah, let's do this. If we put it in a simple model, we can even identify which items to update when we eventually get citations, right? Seems simple enough to me and worth doing?

**Comment by grossir** [comment]
Closing this; we actually finished this specific citation extraction round
